pkg/handler/iam_handler_test.go

```diff
@@ -0,0 +1,565 @@
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "failed_with_projects_server_set_iam_policy_error",
+			orgsServer: &fakeOrgsServer{
+				policy: &iampb.Policy{},
+			},
+			foldersServer: &fakeFoldersServer{
+				policy: &iampb.Policy{},
+			},
+			projectsServer: &fakeProjectsServer{
+				policy:          &iampb.Policy{},
+				setIAMPolicyErr: fmt.Errorf("Set IAM policy encountered error: Internal Server Error"),
+			},
+			request: &v1alpha1.IAMRequestWrapper{
+				IAMRequest: &v1alpha1.IAMRequest{
+					ResourcePolicies: []*v1alpha1.ResourcePolicy{
+						{
+							Resource: "folders/bar",
+							Bindings: []*v1alpha1.Binding{
+								{
+									Members: []string{
+										"user:test-folder-user@example.com",
+									},
+									Role: "roles/cloudkms.cryptoOperator",
+								},
+							},
+						},
+						{
+							Resource: "projects/baz",
+							Bindings: []*v1alpha1.Binding{
+								{
+									Members: []string{
+										"user:test-project-user@example.com",
+									},
+									Role: "roles/bigquery.dataViewer",
+								},
+							},
+						},
+					},
+				},
+				Duration: 1 * time.Hour,
+			},
+			wantPolicies: []*v1alpha1.IAMResponse{
+				{
+					Resource: "folders/bar",
+					Policy: &iampb.Policy{
+						Bindings: []*iampb.Binding{
+							{
+								Members: []string{
+									"user:test-folder-user@example.com",
+								},
+								Role: "roles/cloudkms.cryptoOperator",
+								Condition: &expr.Expr{
+									Title:      ConditionTitle,
+									Expression: fmt.Sprintf("request.time < timestamp('%s')", time.Now().Add(1*time.Hour).Format(time.RFC3339)),
+								},
+							},
+						},
+					},
+				},
+			},
+			wantErrSubstr:  "Set IAM policy encountered error: Internal Server Error",
+			wantOrgsPolicy: &iampb.Policy{},
+			wantFoldersPolicy: &iampb.Policy{
+				Bindings: []*iampb.Binding{
+					{
+						Members: []string{
+							"user:test-folder-user@example.com",
+						},
+						Role: "roles/cloudkms.cryptoOperator",
+						Condition: &expr.Expr{
+							Title:      ConditionTitle,
+							Expression: fmt.Sprintf("request.time < timestamp('%s')", time.Now().Add(1*time.Hour).Format(time.RFC3339)),
+						},
+					},
+				},
+			},
+			wantProjectsPolicy: &iampb.Policy{},
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.Background()
+
+			// Setup fake servers.
+			addrOrg, connOrg := testutil.FakeGRPCServer(t, func(s *grpc.Server) {
+				resourcemanagerpb.RegisterOrganizationsServer(s, tc.orgsServer)
+			})
+			defer connOrg.Close()
+			addrFol, connFol := testutil.FakeGRPCServer(t, func(s *grpc.Server) {
+				resourcemanagerpb.RegisterFoldersServer(s, tc.foldersServer)
+			})
+			defer connFol.Close()
+			addrPro, connPro := testutil.FakeGRPCServer(t, func(s *grpc.Server) {
+				resourcemanagerpb.RegisterProjectsServer(s, tc.projectsServer)
+			})
```

Comment:
Do you need one server for each? Can we just register all services to the same gRPC server?

Reply:
Looks like we can! Thx!